bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -155,9 +145,13 @@ public void initialize() {
 
         configuration = getConfigAs(SamsungTvConfiguration.class);
 
-        discoveryServiceRegistry.addDiscoveryListener(this);
+        // One Samsung TV contains several UPnP devices. Create handler for each UPnP device.
+        for (RemoteDevice device : upnpService.getRegistry().getRemoteDevices()) {
+            remoteDeviceAdded(null, device);
+        }
+        upnpService.getRegistry().addListener(this);
```

Comment:
There might be a small chance to miss device(s) if new is registered between getRemoteDevices loop and addListener.

Reply:
Addressed.